### ISSUE TITLE
Maximize window right buggy

### ISSUE THREAD

**Issue by mruckli** [fifafu/BetterTouchTool#179]
#### Description of your bug/question/feature request...:
When executing the maximize window right action the window gets maximized to the left side. This only happes when a window was maximized to fullscreen when executed. When executing it the second time the resizing works as expected.

See the attached screenshot for the defined gestures.
![screen shot 2016-12-29 at 15 40 43](https://cloud.githubusercontent.com/assets/1864153/21546317/4bf79b46-cdde-11e6-9a0f-421e7f5434d9.png)

Any Ideas?

**What kind of Mac are you using?**
MacBook Pro (Retina, 15-inch, Mid 2015)
2.2 GHz Intel Core i7
16 GB 1600 MHz DDR3

**Which input device (e.g. MacBook Trackad, Magic Mouse, Magic Trackpad Touch Bar) is affected by this?**
External Trackpad
MacBook Trackpad

**Which versions of macOS and BetterTouchTool are you running?**
Sierra: 10.12.2 (16C67)
BTT: 1.978

**Comment by fifafu** [comment]
I can't seem to reproduce this on my machines, do you have a special display setup?

**Comment by mruckli** [comment]
Ah, good input! I just have one cinema display attached. When I unplug it it works like a charm :/

**Comment by fifafu** [comment]
Weird, could you check whether moving da display a little bit up or down in System Preferences => Displays fixes this? (sometimes the system reports wrong values to BTT and this resets that)

**Comment by mruckli** [comment]
I just tried, moved both of them. Did not fix the issue

**Comment by custodie** [comment]
I have the same problem. I have an identical machine configuration, except for running High Sierra (V 10.12.1) instead. 

I'm running BTT version 2.315 but I've had this problem for a long time, no matter the BTT version. Unplugging all my peripherals, including the monitors) doesn't fix the problem. 

I noticed the problem happens with some applications and not with others. For example, BTT doesn't maximize Safari windows (currently running version 11.01). It also doesn't maximize outlook windows (currently running version 15.40). Instead, the windows inch to the with each attempt until it reaches the edge of the monitor, at which point it maximizes by expanding all the way to the right (as intended). So it seems as though the maximizing the window only works if the target window is expanding to their righthand side. 

The feature does work with a few application such as Powerpoint (Version 15.40) and native mac OS windows, such as Finder windows. 

My keyboard shortcut is as follows but I've tried other peripheral such as the magic mouse and trackpad
Keyboard Shortcut, ⌘⇧↑: Maximize Window

I hope this helps in replicating the problem and finding a fix as this is a very painful bug. 

Thank you for creating such a helpful tool.
